source/slang/slang-emit.cpp

```diff
@@ -362,6 +362,42 @@ void calcRequiredLoweringPassSet(RequiredLoweringPassSet& result, CodeGenContext
     }
 }
 
+void checkStaticAssert(CodeGenContext* codeGenContext, IRInst* inst, DiagnosticSink* sink)
+{
+    switch (inst->getOp())
+    {
+    case kIROp_StaticAssert:
+    {
+        IRInst* condi = inst->getOperand(0);
+        if (auto condiLit = as<IRBoolLit>(condi))
+        {
+            if (!condiLit->getValue())
+            {
+                IRInst* msg = inst->getOperand(1);
+                if (auto msgLit = as<IRStringLit>(msg))
+                {
+                    sink->diagnose(inst, Diagnostics::staticAssertionFailure, msgLit->getStringSlice());
+                }
+                else
+                {
+                    sink->diagnose(inst, Diagnostics::staticAssertionFailureWithoutMessage);
+                }
+            }
+        }
+        else
+        {
+            sink->diagnose(condi, Diagnostics::staticAssertionConditionNotConstant);
+        }
+        break;
```

Comment:
You can just `inst->removeAndDeallocate()` here to remove the static assert inst since validation is already done at this point and the StaticAssert inst has no value past this point.

Reply:
This worked!

Reply:
Ok. I turned out that it causes a problem if I remove as suggested.
When I remove StaticAssert after it is applied once, it doesn't work for the next specialization.

Reply:
The idea is to apply the check only once when we know we have done everything we can to resolve  the values, so we shouldn't be running the checking pass more than once.

Reply:
I am gonna need to go over one of the brownbag records and understand better how the generics are instantiated. But my understanding here is that the static assertion has to  be alive for the next specialization if the StaticAssert was in a generic function.

For the cases where the functions are not generic, I guess I can removed them.
 And I can probably remove StaticAssert from specialized functions; not from the generic functions.
I will try more on this.